Versioning.md

```diff
@@ -0,0 +1,38 @@
+# iTwin.js versioning policies
+
+iTwin.js employs versioning policies to faciliate the following objectives:
+
+- Provide a broad set of stable, well-supported APIs.
+- Continuously deliver new features, bug fixes, and security updates.
+- Evolve APIs over time in response to user feedback.
+- Encourage users to keep up to date with the most recent versions.
+
+## API promotion process
+
+Each API exported by a package is marked with a [release tag](./docs/learning/guidelines/release-tags-guidelines.md) indicating its level of stability. A package's "public" API comprises all of its exported APIs that have been marked with the `@public` release tag. All other APIs are considered "non-public". The [package versioning](#package-versioning-policy) and [API deprecation](#api-deprecation-policy) policies apply only the a package's public API.
+
+`@internal` APIs are never intended to be used outside of the iTwin.js library itself. All other APIs are intended to ultimately become `@public` through the following process:
+
+- A highly experimental API may start out tagged as `@alpha`, indicating it is not yet mature enough for users to try out. This stage is usually skipped.
+- A new API, or one that has been promoted from `@alpha`, is tagged as `@beta` to invite users to experiment with it and provide feedback. The API may change in response to feedback from one release to the next.
+  - Note: extension APIs are tagged as `@preview` instead of `@beta`.
+- A `@beta` API that has been sufficiently tested is promoted to `@public` to indicate that it is now subject to [stability guarantees](#package-versioning-policy).
+- A `@public` API may be tagged as `@deprecated` indicating that it may later be removed according to the [deprecation policy](#api-deprecation-policy).
+
+Occasionally, a `@beta` or `@alpha` API may be abandoned rather than being promoted. Not all experiments are successful.
+
+In practice, `@beta` and `@public` APIs are typically designed in a way that promotes flexibility in the future - e.g., by accepting their arguments as "options" object types. This reduces the risk of breaking changes and deprecation.
+
+## Package versioning policy
+
+The individual packages that constitute the iTwin.js libraries use [semantic versioning](./docs/learning/guidelines/semantic-versioning-guidelines.md) to convey information about how their public APIs change over time.
+
+- Patch releases occur frequently to fix bugs or security issues and have no impact on the public API.
+- Minor releases occur on roughly six- to eight-week intervals. They may introduce new APIs, but never break any existing `@public` APIs.
+- Major releases occur only as frequently as necessary. They can include breaking changes to `@public` APIs, but only according to the [deprecation policy](#api-deprecation-policy).
+
```

Comment:
@calebmshafer I would like to link to (or include?) the LTS major release support policy here.

Reply:
@calebmshafer separate PR?

Reply:
Found https://github.com/iTwin/itwinjs-backlog/issues/391 where you were working on LTS policy, I guess I will fold it in here...

Reply:
IMO semantic-versioning-guidelines.md should go away entirely. It mostly just restates how semver works - a link to somebody else's description would suffice.

release-tags-guidelines.md should tell contributors to iTwin.js how to use release tags. The info in Versioning.md is targeted at consumers of iTwin.js, not its developers. That's why it's not in the docs directory.

Are you asking me to clean both of those other documents up as part of this PR; or, remove references to them from this document; or something else?

Reply:
> Are you asking me to clean both of those other documents up as part of this PR; or, remove references to them from this document; or something else?

I was referring to removing the semantic versioning one. Just pushed an update to do that.

> The info in Versioning.md is targeted at consumers of iTwin.js, not its developers. That's why it's not in the docs directory.

If that's the goal, I think this one is too developer focused. The other one I'm working on that's the support policy would be the high level item that both dev and non-dev would be looking at IMO. In both cases, I feel like we should have them in our deployed docs site as well as referenced in the README.

Reply:
> I think this one is too developer focused

If you want me to do something about that I'll need you to elaborate. ("Consumers" here means, developers who **use** iTwin.js, but do not **contribute** to iTwin.js).

Reply:
Yes, I meant the developer using iTwin.js, not developing it, as well. If that's the goal of this document why wouldn't it be on our doc site? I think it'd be helpful to have it there and then also keep the link that you added in our main README.